src/Microsoft.AspNet.OData.Shared/DataModificationExceptionType.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.OData.Builder;
+
+namespace Org.OData.Core.V1
+{
+    /// <summary>
+    /// Represents a Message Type
+    /// </summary>
+    public class MessageType
+    {
+        /// <summary>
+        /// Code of message
+        /// </summary>
+        public string Code { get; set; }
+
+        /// <summary>
+        /// Actual message
+        /// </summary>
+        public string Message { get; set; }
+
+        /// <summary>
+        /// Severity of message
+        /// </summary>
+        public string Severity { get; set; }
+
+        /// <summary>
+        /// Target of message
+        /// </summary>
+        public string Target { get; set; }
+
+        /// <summary>
+        /// Details of message
+        /// </summary>
+        public string Details { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an Exception Type
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1012:AbstractTypesShouldNotHaveConstructors")]
+    public abstract class ExceptionType
+    {
+        /// <summary>
+        /// Instanciates a type of <see cref="ExceptionType"/>
+        /// </summary>
+        public ExceptionType()
+        {
+
+        }
+
+        /// <summary>
+        /// Represents a MessageType
+        /// </summary>
+        public MessageType MessageType { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an Exception for Data modification Operation.
+    /// </summary>    
+    public class DataModificationExceptionType : ExceptionType
+    {
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="DataModificationExceptionType"/> class.
+        /// </summary>
+        public DataModificationExceptionType(DataModificationOperationKind failedOperation)
+        {
+            this.FailedOperation = failedOperation;
+        }
+
+        /// <summary>
+        /// Represents king of <see cref="DataModificationOperationKind"/> type of operation 
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1709:IdentifiersShouldBeCasedCorrectly", MessageId = "failed")]
+        public DataModificationOperationKind FailedOperation { get; }
+
+        /// <summary>
+        /// Represents response code
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1709:IdentifiersShouldBeCasedCorrectly", MessageId = "response")]
+        public Int16 ResponseCode { get; set; }
```

Comment:
Is there any _tangible_ benefit of using `Int16` instead of just `int`?

Reply:
we are getting it from vocabs